bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/component/Vacuum.java

```diff
@@ -0,0 +1,264 @@
+import java.util.List;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.mqtt.generic.values.DateTimeValue;
+import org.openhab.binding.mqtt.generic.values.NumberValue;
+import org.openhab.binding.mqtt.generic.values.TextValue;
+import org.openhab.binding.mqtt.homeassistant.internal.config.dto.AbstractChannelConfiguration;
+
+/**
+ * A MQTT vacuum, following the https://www.home-assistant.io/components/vacuum.mqtt/ specification.
+ *
+ * @author Stefan Triller - Initial contribution
+ */
+@NonNullByDefault
+public class Vacuum extends AbstractComponent<Vacuum.ChannelConfiguration> {
+    public static final String VACUUM_STATE_CHANNEL_ID = "state";
+    public static final String VACUUM_COMMAND_CHANNEL_ID = "command";
+    public static final String VACUUM_BATTERY_CHANNEL_ID = "batteryLevel";
+    public static final String VACUUM_FAN_SPEED_CHANNEL_ID = "fanSpeed";
+
+    // sensor stats
+    public static final String VACUUM_MAIN_BRUSH_CHANNEL_ID = "mainBrushUsage";
+    public static final String VACUUM_SIDE_BRUSH_CHANNEL_ID = "sideBrushUsage";
+    public static final String VACUUM_FILTER_CHANNEL_ID = "filter";
+    public static final String VACUUM_SENSOR_CHANNEL_ID = "sensor";
+    public static final String VACUUM_CURRENT_CLEAN_TIME_CHANNEL_ID = "currentCleanTime";
+    public static final String VACUUM_CURRENT_CLEAN_AREA_CHANNEL_ID = "currentCleanArea";
+    public static final String VACUUM_CLEAN_TIME_CHANNEL_ID = "cleanTime";
+    public static final String VACUUM_CLEAN_AREA_CHANNEL_ID = "cleanArea";
+    public static final String VACUUM_CLEAN_COUNT_CHANNEL_ID = "cleanCount";
+
+    public static final String VACUUM_LAST_RUN_START_CHANNEL_ID = "lastRunStart";
+    public static final String VACUUM_LAST_RUN_END_CHANNEL_ID = "lastRunEnd";
+    public static final String VACUUM_LAST_RUN_DURATION_CHANNEL_ID = "lastRunDuration";
+    public static final String VACUUM_LAST_RUN_AREA_CHANNEL_ID = "lastRunArea";
+    public static final String VACUUM_LAST_RUN_ERROR_CODE_CHANNEL_ID = "lastRunErrorCode";
+    public static final String VACUUM_LAST_RUN_ERROR_DESCRIPTION_CHANNEL_ID = "lastRunErrorDescription";
+    public static final String VACUUM_LAST_RUN_FINISHED_FLAG_CHANNEL_ID = "lastRunFinishedFlag";
+
+    public static final String VACUUM_BIN_IN_TIME_CHANNEL_ID = "binInTime";
+    public static final String VACUUM_LAST_BIN_OUT_TIME_CHANNEL_ID = "lastBinOutTime";
+    public static final String VACUUM_LAST_BIN_FULL_TIME_CHANNEL_ID = "lastBinFullTime";
+
+    public static final String VACUUM_CUSMTOM_COMMAND_CHANNEL_ID = "customCommand";
+
+    /**
+     * Configuration class for MQTT component
+     */
+    static class ChannelConfiguration extends AbstractChannelConfiguration {
+        ChannelConfiguration() {
+            super("MQTT Vacuum");
+        }
+
+        protected @Nullable String commandTopic;
+        protected String stateTopic = "";
```

Comment:
Does this work?
Don't we need a @SerializedName("state_topic") vor the deserialization to work?